simulation.cpp

```diff
@@ -397,30 +397,32 @@ void test_simulation() noexcept//!OCLINT test may be many
         assert(are_same_env_functions(identity_env.env_function_A, identity));
 
         auto identity_net = net_param{{1,1}, linear};
-        assert(net_behaves_like_the_function(identity_net, identity));
+        assert(net_behaves_like_the_function(change_all_weights(identity_net, 1), identity));
 
         auto identity_ind = ind_param{identity_net};
 
         int pop_size = 2;
         auto minimal_pop = pop_param{pop_size, 0, 0};
 
+        auto sim_p = sim_param{};
+        sim_p.selection_strength = 2;
+
 
         simulation s{all_params{
                 identity_env,
                 identity_ind,
                 minimal_pop,
-                sim_param{}
+                sim_p
```

Comment:
The problem doing that was the selection strength initialized at 0, which made the whole thing not work because you cannot test if selection works when there is no selection!